plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/actions/refresh-button.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import RefreshIcon from '../../../../assets/images/refresh.svg';
+import { SubscriptionsContext } from '../../../../contexts/subscriptions-context';
+import { addNotice, removeNotice } from '../../../../utils/functions';
+import { NoticeStatus } from '../../../../contexts/types';
+
+const NOTICE_ID = 'woocommerce-marketplace-refresh-subscriptions';
+
+export function RefreshButton() {
+	const { refreshSubscriptions } = useContext( SubscriptionsContext );
+	const [ isLoading, setIsLoading ] = useState( false );
+
+	const refresh = () => {
+		if ( isLoading ) {
+			return;
+		}
+
+		removeNotice( NOTICE_ID );
+		setIsLoading( true );
+
+		refreshSubscriptions()
+			.then( () => {
+				addNotice(
+					NOTICE_ID,
+					__( 'Subscriptions refreshed.', 'woocommerce' ),
+					NoticeStatus.Success
+				);
+			} )
+			.catch( ( error ) => {
+				addNotice(
+					NOTICE_ID,
+					sprintf(
+						// translators: %s is the error message.
+						__(
+							'Error refreshing subscriptions: %s',
+							'woocommerce'
+						),
+						error.message
+					),
+					NoticeStatus.Error,
+					{
+						actions: [
+							{
+								label: __( 'Try again', 'woocommerce' ),
+								onClick: refresh,
+							},
+						],
+					}
+				);
+			} )
+			.finally( () => {
+				setIsLoading( false );
+			} );
+	};
+
+	return (
+		<Button
+			className="woocommerce-marketplace__refresh-subscriptions"
+			onClick={ refresh }
+			isBusy={ isLoading }
+		>
+			<img
+				src={ RefreshIcon }
+				alt={ __( 'Refresh subscriptions', 'woocommerce' ) }
+				className="woocommerce-marketplace__refresh-subscriptions__icon"
```

Comment:
We're stretching the `block__element--modifier` format rather with these classnames. I know we already have `woocommerce-marketplace__my-subscriptions__installed` and so on, but I would try and at least limit the mutations by using `woocommerce-marketplace__refresh-subscriptions-icon` here.